tools/Voron.Dictionary.Generator/Program.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Text;
+using Sparrow;
+using Sparrow.Server.Compression;
+using Voron.Data;
+using Voron.Data.CompactTrees;
+using Voron.Global;
+
+int pagesToUse = 8;
+int tableSize = pagesToUse * Constants.Storage.PageSize - PageHeader.SizeOf - PersistentDictionaryHeader.SizeOf;
+
+var data = new byte[tableSize];
+unsafe
+{
+    // Find all .txt files in the current directory
+    var files = Directory.GetFiles(Directory.GetCurrentDirectory(), "*.txt");
+
+    var dictionary = new List<string>();
+    foreach (var file in files)
+    {
+        foreach (var line in File.ReadAllLines(file))
+            dictionary.Add(line);
+    }
+
+    fixed (byte* dataPtr = data)
+    {
+        var encoder = new HopeEncoder<Encoder3Gram<NativeMemoryEncoderState>>(
+        new Encoder3Gram<NativeMemoryEncoderState>(
+            new NativeMemoryEncoderState(dataPtr, tableSize)));
+
+        int lowerBound = 128;
+        int higherBound = short.MaxValue - 1;
+
+        while (lowerBound < higherBound)
+        {
+            int current = (lowerBound + higherBound) / 2;
+            if (current == lowerBound)
+                break;
+            
+            try
+            {
+                encoder.Train(new StringArrayIterator(dictionary.ToArray()), current);
+                lowerBound = current;
+            }
+            catch
+            {
+                higherBound = current-1;
+                continue;
+            }
+        }
+
+        var treesDirectory = new DirectoryInfo("..\\..\\..\\..\\..\\src\\Voron\\Data\\CompactTrees");
+
+        var dictionarySize = lowerBound;
+        encoder.Train(new StringArrayIterator(dictionary.ToArray()), dictionarySize);
```

Comment:
This is how we train the dictionary and just dump the table, as we do in Voron (using the same format).